Mlem/Models/Content/Community/Community List/CommunityListModel.swift

```diff
@@ -205,28 +116,89 @@ class CommunityListModel: ObservableObject {
+            self.visibleSections = newVisibleSections
         }
     }
     
-    private func updateFavorites(_ favorites: [APICommunity]) {
-        combine(subscriptions, favorites)
+    private func recomputeSections() -> (all: [CommunityListSection], visible: [CommunityListSection]) {
+        var newAllSections: [CommunityListSection] = .init()
+        var newVisibleSections: [CommunityListSection] = .init()
+        
+        let topSection = withDependencies(from: self) {
+            CommunityListSection(
+                viewId: "top",
+                sidebarEntry: .init(sidebarLabel: nil, sidebarIcon: "line.3.horizontal"),
+                inlineHeaderLabel: nil,
+                accessibilityLabel: "Top of communities",
+                communities: .init()
+            )
+        }
+        newAllSections.append(topSection)
+        
+        let favoritesSection = withDependencies(from: self) {
+            CommunityListSection(
+                viewId: "favorites",
+                sidebarEntry: .init(sidebarLabel: nil, sidebarIcon: "star.fill"),
+                inlineHeaderLabel: "Favorites",
+                accessibilityLabel: "Favorited Communities",
+                communities: favorited
+            )
+        }
+        newAllSections.append(favoritesSection)
+        if !favorited.isEmpty {
+            newVisibleSections.append(favoritesSection)
+        }
+        
+        let alphabeticSections = alphabeticSections()
+        
+        newAllSections.append(contentsOf: alphabeticSections)
+        newVisibleSections.append(contentsOf: alphabeticSections.filter { section in
+            !section.communities.isEmpty
+        })
+        
+        return (all: newAllSections, visible: newVisibleSections)
     }
     
-    private func combine(_ subscriptions: [APICommunity], _ favorites: [APICommunity]) {
-        // store the values for future use...
-        self.subscriptions = subscriptions
-        favoriteCommunities = favorites
+    private func alphabeticSections() -> [CommunityListSection] {
+        let sections: [String: [APICommunity]] = .init(
+            grouping: subscribed,
+            by: { item in
+                if let first = item.name.first, first.isLetter {
+                    return first.uppercased()
+                }
+                return "#"
+            }
+        )
         
-        // combine and sort the two lists, excluding duplicates
-        let combined = subscriptions + favorites.filter { !subscriptions.contains($0) }
-        let sorted = combined.sorted()
+        assert(sections.values.reduce(0) { x, communities in
+            x + communities.count
+        } == subscribed.count, "mapping operation produced mismatched counts")
         
-        // update our published value for the view to render
-        mainQueue.schedule { [weak self] in
-            self?.communities = sorted
+        return [String].labelAlphabet.map { character in
```

Comment:
Using `labelAlphabet` here means that communities that begin with an accented character (`à`, etc) won't be shown in the list. `Character.isLetter`, which we're using in the grouping logic, returns `true` for these characters. Lemmy won't let you create communities with those characters in the name, but KBin does. 

To fix this, we could either:
- Use `sections.keys.map` here instead.
- Modify the grouping logic to use `[String].labelAlphabet.contains()` instead of `isLetter`.
- Modify the grouping logic to categorize accented characters under their base version. This could get dicey for non-latin alphabets, though.

Reply:
Nice catch 👍 

`sections.keys.sorted().map` seems like the best approach--we lose a shred of time needing to sort it, but it's only ~28 items so that shouldn't matter.